sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiverAsyncClient.java

```diff
@@ -0,0 +1,124 @@
+    private final ClientLogger logger = new ClientLogger(ServiceBusSessionReceiverAsyncClient.class);
+
+    ServiceBusSessionReceiverAsyncClient(String fullyQualifiedNamespace, String entityPath,
+        MessagingEntityType entityType, ReceiverOptions receiverOptions,
+        ServiceBusConnectionProcessor connectionProcessor, TracerProvider tracerProvider,
+        MessageSerializer messageSerializer, Runnable onClientClose) {
+        this.fullyQualifiedNamespace = Objects.requireNonNull(fullyQualifiedNamespace,
+            "'fullyQualifiedNamespace' cannot be null.");
+        this.entityPath = Objects.requireNonNull(entityPath, "'entityPath' cannot be null.");
+        this.entityType = Objects.requireNonNull(entityType, "'entityType' cannot be null.");
+        this.receiverOptions = Objects.requireNonNull(receiverOptions, "'receiveOptions cannot be null.'");
+        this.connectionProcessor = Objects.requireNonNull(connectionProcessor, "'connectionProcessor' cannot be null.");
+        this.tracerProvider = Objects.requireNonNull(tracerProvider, "'tracerProvider' cannot be null.");
+        this.messageSerializer = Objects.requireNonNull(messageSerializer, "'messageSerializer' cannot be null.");
+        this.onClientClose = Objects.requireNonNull(onClientClose, "'onClientClose' cannot be null.");
+        this.unNamedSessionManager = new ServiceBusSessionManager(entityPath, entityType,
+            connectionProcessor, tracerProvider,
+            messageSerializer, receiverOptions);
+    }
+
+    /**
+     * Create a link for the next available session and use the link to create a {@link ServiceBusReceiverAsyncClient}
+     * to receive messages from that session.
+     * @return A {@link ServiceBusReceiverAsyncClient} that is tied to the available session.
+     */
+    public Mono<ServiceBusReceiverAsyncClient> acceptNextSession() {
+        return unNamedSessionManager.getActiveLink().flatMap(receiveLink -> receiveLink.getSessionId()
+            .map(sessionId -> {
+                ReceiverOptions newReceiverOptions = new ReceiverOptions(receiverOptions.getReceiveMode(),
+                    receiverOptions.getPrefetchCount(), sessionId, null, receiverOptions.getMaxLockRenewDuration());
+                final ServiceBusSessionManager sessionSpecificManager = new ServiceBusSessionManager(entityPath,
+                    entityType, connectionProcessor, tracerProvider, messageSerializer, newReceiverOptions,
+                    receiveLink);
+                return new ServiceBusReceiverAsyncClient(fullyQualifiedNamespace, entityPath,
+                    entityType, newReceiverOptions, connectionProcessor, ServiceBusConstants.OPERATION_TIMEOUT,
+                    tracerProvider, messageSerializer, onClientClose, sessionSpecificManager);
+            }));
+    }
+
+    /**
+     * Create a link for the "sessionId" and use the link to create a {@link ServiceBusReceiverAsyncClient}
+     * to receive messages from the session.
```

Comment:
Add documentation to specify what happens if the requested sessionId is not available.

Reply:
Added a description